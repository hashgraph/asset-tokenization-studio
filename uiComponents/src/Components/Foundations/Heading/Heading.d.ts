import type { BoxProps as ChakraBoxProps, HeadingProps as ChakraHeadingProps } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
export interface HeadingProps extends ChakraBoxProps {
    size?: ChakraHeadingProps["size"];
    variant?: ChakraHeadingProps["variant"];
}
export declare const Heading: ComponentWithAs<"h2", HeadingProps>;
