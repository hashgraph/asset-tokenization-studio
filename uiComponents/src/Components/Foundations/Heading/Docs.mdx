import {
  Story,
  Meta,
  Title,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as HeadingStories from "./Heading.stories";

<Meta of={HeadingStories} />

<Title />
<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A20660&t=gryWgf3cMhWIZY1U-4"
  target="\_blank"
>
  View figma design
</a>
<Primary />

## Props

<ArgTypes of={HeadingStories} />

## Stories

<Stories title="" />
