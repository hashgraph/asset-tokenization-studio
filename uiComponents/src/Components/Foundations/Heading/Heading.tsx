import type {
  BoxProps as ChakraBoxProps,
  HeadingProps as ChakraHeadingProps,
} from "@chakra-ui/layout";
import { Box as ChakraBox } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
import {
  forwardRef,
  useStyleConfig as useChakraStyleConfig,
} from "@chakra-ui/system";
import React from "react";

export interface HeadingProps extends ChakraBoxProps {
  size?: ChakraHeadingProps["size"];
  variant?: ChakraHeadingProps["variant"];
}

export const Heading: ComponentWithAs<"h2", HeadingProps> = forwardRef<
  HeadingProps,
  "h2"
>(
  (
    { children, as = "h2", size, variant, textStyle, ...props }: HeadingProps,
    ref
  ) => {
    const styles = useChakraStyleConfig("Heading", { size, variant });

    return (
      <ChakraBox
        ref={ref}
        as={as}
        textStyle={textStyle}
        {...props}
        {...(!textStyle && { __css: styles })}
      >
        {children}
      </ChakraBox>
    );
  }
);
