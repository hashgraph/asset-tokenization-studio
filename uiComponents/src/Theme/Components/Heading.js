export var ConfigHeading = {
    baseStyle: {
        color: "black"
    }
};
