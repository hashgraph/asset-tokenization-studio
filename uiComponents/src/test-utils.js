import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
