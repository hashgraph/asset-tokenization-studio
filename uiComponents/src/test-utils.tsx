import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
