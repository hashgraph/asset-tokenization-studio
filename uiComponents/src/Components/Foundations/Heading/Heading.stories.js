var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { ThemeStoryWrapper } from "../../story-config";
import { Heading } from "./Heading";
import { Title } from "@storybook/addon-docs";
import { Box, Grid, GridItem, Link } from "@chakra-ui/react";
import { textStyles } from "@/Theme/textStyles";
var headingTextStyles = Object.keys(textStyles).filter(function (textStyle) {
    return textStyle.startsWith("Heading");
});
var meta = {
    title: "Design System/Foundations/Heading",
    component: Heading,
    argTypes: {
        textStyle: {
            options: headingTextStyles,
            control: { type: "select" },
            description: "The textStyle that the heading has. Must be defined in the theme in the theme key `textStyles` "
        },
        size: {
            description: "The size of the text. We recommend using `textStyle` instead. "
        },
        variant: {
            description: "The variant of the text. Must be defined in the theme inside `Text` component."
        }
    },
    parameters: {
        docs: {
            page: function () { return (React.createElement(ThemeStoryWrapper, null,
                React.createElement(React.Fragment, null,
                    React.createElement(Title, null, "Heading Text styles"),
                    React.createElement(Grid, { templateRows: "repeat(4, 1fr)", gap: 10, mt: 10, alignItems: "center", gridAutoFlow: "column", justifyItems: "center", bg: "neutral" }, headingTextStyles.map(function (name) { return (React.createElement(GridItem, { key: name },
                        React.createElement(Heading, { textStyle: name }, name))); })),
                    React.createElement(Box, { mt: 10 },
                        React.createElement(Link, { href: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A20660&t=gryWgf3cMhWIZY1U-4", isExternal: true }, "View figma design"))))); }
        }
    },
    args: {}
};
export default meta;
var Template = function (args) { return (React.createElement(Heading, __assign({}, args), "Title")); };
export var Default = Template.bind({});
Default.args = {};
