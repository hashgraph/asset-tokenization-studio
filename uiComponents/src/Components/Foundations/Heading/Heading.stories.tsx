import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { ThemeStoryWrapper } from "../../story-config";
import { Heading } from "./Heading";
import { Title } from "@storybook/addon-docs";
import { Box, Grid, GridItem, Link } from "@chakra-ui/react";
import { textStyles } from "@/Theme/textStyles";

const headingTextStyles = Object.keys(textStyles).filter((textStyle) =>
  textStyle.startsWith("Heading")
);

const meta = {
  title: "Design System/Foundations/Heading",
  component: Heading,
  argTypes: {
    textStyle: {
      options: headingTextStyles,
      control: { type: "select" },
      description:
        "The textStyle that the heading has. Must be defined in the theme in the theme key `textStyles` ",
    },
    size: {
      description:
        "The size of the text. We recommend using `textStyle` instead. ",
    },
    variant: {
      description:
        "The variant of the text. Must be defined in the theme inside `Text` component.",
    },
  },
  parameters: {
    docs: {
      page: () => (
        <ThemeStoryWrapper>
          <>
            <Title>Heading Text styles</Title>

            <Grid
              templateRows="repeat(4, 1fr)"
              gap={10}
              mt={10}
              alignItems="center"
              gridAutoFlow={"column"}
              justifyItems="center"
              bg="neutral"
            >
              {headingTextStyles.map((name) => (
                <GridItem key={name}>
                  <Heading textStyle={name}>{name}</Heading>
                </GridItem>
              ))}
            </Grid>

            <Box mt={10}>
              <Link
                href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A20660&t=gryWgf3cMhWIZY1U-4"
                isExternal
              >
                View figma design
              </Link>
            </Box>
          </>
        </ThemeStoryWrapper>
      ),
    },
  },
  args: {},
} as Meta<typeof Heading>;
export default meta;

const Template: StoryFn<typeof Heading> = (args) => (
  <Heading {...args}>Title</Heading>
);

export const Default = Template.bind({});
Default.args = {};
