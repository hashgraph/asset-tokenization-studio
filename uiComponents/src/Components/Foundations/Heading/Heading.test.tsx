import React from "react";
import { render } from "../../../test-utils";
import type { HeadingProps } from "./Heading";
import { Heading } from "./Heading";

describe(`<Heading />`, () => {
  const defaultProps = {
    fontSize: "32px",
  };

  const themeCustom = {
    textStyles: {
      red: {
        color: "red",
      },
    },
    components: {
      Heading: {
        variants: {
          light: {
            fontFamily: "Helvetica Neue",
          },
        },
      },
    },
  };

  const factoryComponent = (props: HeadingProps = defaultProps) =>
    render(
      <Heading {...props} data-testid="heading">
        Title
      </Heading>,
      themeCustom
    );

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders correctly with text style", () => {
    const component = factoryComponent({ textStyle: "red" });

    expect(component.asFragment()).toMatchSnapshot("WithTextStyle");
  });

  test("renders correctly with variant", () => {
    const component = factoryComponent({ variant: "light" });

    expect(component.asFragment()).toMatchSnapshot("WithVariant");
  });

  test("renders correctly with custom heading tag", () => {
    const component = factoryComponent({ as: "h1" });

    expect(component.getByTestId("heading").tagName).toBe("H1");
    expect(component.asFragment()).toMatchSnapshot("WithCustomTag");
  });
});
