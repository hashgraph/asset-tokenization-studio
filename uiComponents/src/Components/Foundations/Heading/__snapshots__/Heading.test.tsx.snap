// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders correctly 1`] = `
<DocumentFragment>
  <h2
    class="css-idm3bz"
    data-testid="heading"
  >
    Title
  </h2>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Heading /> renders correctly with custom heading tag: WithCustomTag 1`] = `
<DocumentFragment>
  <h1
    class="css-0"
    data-testid="heading"
  >
    Title
  </h1>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Heading /> renders correctly with text style: WithTextStyle 1`] = `
<DocumentFragment>
  <h2
    class="css-tokvmb"
    data-testid="heading"
  >
    Title
  </h2>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Heading /> renders correctly with variant: WithVariant 1`] = `
<DocumentFragment>
  <h2
    class="css-0"
    data-testid="heading"
  >
    Title
  </h2>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
